File: backend/src/models/Appointment.js
const mongoose = require("mongoose");

const appointmentSchema = new mongoose.Schema({
  userId: { type: mongoose.Schema.Types.ObjectId, ref: "User" },
  hospitalId: { type: mongoose.Schema.Types.ObjectId, ref: "Hospital" },
  vaccineId: { type: mongoose.Schema.Types.ObjectId, ref: "Vaccine" },
  date: Date,
  status: { type: String, enum: ["pending", "approved", "completed"], default: "pending" },
  qrCode: String,
});

module.exports = mongoose.model("Appointment", appointmentSchema);

File: backend/src/main/java/com/vaccination/model/Appointment.java
package com.vaccination.model;
import jakarta.persistence.*;
import java.time.LocalDateTime;

@Entity
@Table(name="appointments")
public class Appointment {
  @Id @GeneratedValue(strategy = GenerationType.IDENTITY) private Long id;

  @ManyToOne @JoinColumn(name="user_id") private User user;
  @ManyToOne @JoinColumn(name="hospital_id") private Hospital hospital;
  @ManyToOne @JoinColumn(name="vaccine_id") private Vaccine vaccine;

  private LocalDateTime appointmentDate;
  private String status = "PENDING";
  @Lob private String qrCode;
  // getters/setters
}
